app/models/host/discovered.rb

```diff
@@ -86,9 +95,27 @@ def populate_fields_from_facts facts = self.facts_hash, type = 'puppet'
       importer = super(facts)
     end
     self.subnet = Subnet.subnet_for(importer.ip)
+    self.discovery_attribute_set = DiscoveryAttributeSet.where(:host_id => id).first_or_create()
+    self.discovery_attribute_set.update_attributes(import_from_facts())
```

Comment:
leftovers from refactoring the method to not receive arguments, I'll remove them.